app/position/tracking/positiontrackingmanager.cpp

```diff
@@ -0,0 +1,287 @@
+{
+  QSettings settings;
+  settings.remove( QStringLiteral( "MerginMaps/Tracking/TrackedPath" ) );
+}
+
+AbstractTrackingBackend *PositionTrackingManager::constructTrackingBackend( QgsProject *project, PositionKit *positionKit )
+{
+  AbstractTrackingBackend *positionBackend = nullptr;
+  AbstractTrackingBackend::UpdateFrequency frequency = AbstractTrackingBackend::Normal;
+
+  if ( project )
+  {
+    int readFrequency = project->readNumEntry( QStringLiteral( "Mergin" ), QStringLiteral( "PositionTracking/UpdateFrequency" ), 0 );
+
+    if ( readFrequency == 0 )
+    {
+      frequency = AbstractTrackingBackend::Often;
+    }
+    else if ( readFrequency == 1 )
+    {
+      frequency = AbstractTrackingBackend::Normal;
+    }
+    else
+    {
+      frequency = AbstractTrackingBackend::Occasional;
+    }
+  }
+
+  // TODO: android provider
+
+#ifdef Q_OS_IOS
+  positionBackend = new IOSTrackingBackend( frequency );
+#else
+  // desktop
+  if ( positionKit )
+  {
+    positionBackend = new InternalTrackingBackend( positionKit, frequency );
+  }
+  else
+  {
+    CoreUtils::log(
+      QStringLiteral( "Position tracking" ),
+      QStringLiteral( "Requested tracking on desktop, but no position kit provided" )
+    );
+  }
+#endif
+
+  QQmlEngine::setObjectOwnership( positionBackend, QQmlEngine::CppOwnership );
+
+  return positionBackend;
+}
+
+QString PositionTrackingManager::layerId() const
+{
+  return mLayerId;
+}
+
+void PositionTrackingManager::setLayerId( QString newLayerId )
+{
+  if ( mLayerId == newLayerId )
+    return;
+  mLayerId = newLayerId;
+  emit layerIdChanged( mLayerId );
+}
+
+QgsGeometry PositionTrackingManager::trackedGeometry() const
+{
+  return mTrackedGeometry;
+}
+
+AbstractTrackingBackend *PositionTrackingManager::trackingBackend() const
+{
+  return mTrackingBackend.get();
+}
+
+void PositionTrackingManager::setTrackingBackend( AbstractTrackingBackend *newTrackingBackend )
+{
+  if ( mTrackingBackend )
+  {
+    disconnect( mTrackingBackend.get() );
+  }
+
+  mTrackingBackend.reset( newTrackingBackend );
+
+  emit trackingBackendChanged( mTrackingBackend.get() );
+
+  setup();
+}
+
+QgsProject *PositionTrackingManager::qgsProject() const
+{
+  return mQgsProject;
+}
+
+void PositionTrackingManager::setQgsProject( QgsProject *newQgsProject )
+{
+  if ( mQgsProject == newQgsProject )
+    return;
+  mQgsProject = newQgsProject;
+  emit qgsProjectChanged( mQgsProject );
+
+  setup();
+}
+
+VariablesManager *PositionTrackingManager::variablesManager() const
+{
+  return mVariablesManager;
+}
+
+void PositionTrackingManager::setVariablesManager( VariablesManager *newVariablesManager )
+{
+  if ( mVariablesManager == newVariablesManager )
+    return;
+  mVariablesManager = newVariablesManager;
+  emit variablesManagerChanged( mVariablesManager );
+}
+
+void PositionTrackingManager::projectReloaded()
+{
+  // clear pointers to previously loaded project, but keep the geometry
+  mQgsProject = nullptr;
```

Comment:
I am not sure if we should keep the geometry -> isn't better approach when project is about to be reloaded for some different one to flush the tracking to tracking layer of the old project and start new track for the newly reloaded project?

Reply:
Well, reload might happen even when user does not change a project, but syncs with schema change. In this case we would falsely store the tracked path and fill the "tracking_end_time" variable. Instead, we store the tracking path in memory and commit it only when the user decides to stop the tracking (or another project is opened by user).
Moreover, this function was not used at all so I removed it (the pointer is the same as we have the QgsProject singleton).